### PR TITLE
Service page

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

File: test/features/10-functional/10-basic/40-service-details.feature [hunk 1]
```diff
@@ -12,3 +12,20 @@ Feature: Service details page
...
+        When I visit /service/866464
+        Then I should see the phone numbers
+        -------------------------------------------
+        Phone (03) 9418 7447
+        -------------------------------------------
+        When I click on "Other contact options"
+        Then I should see the phone numbers
+        -------------------------------------------
+        Phone (03) 9418 7447
+        Phone 1300 306 645
```
Comment (danni) [inline test/features/10-functional/10-basic/40-service-details.feature]: Can we make this a test for phone type priority (i.e. freecall > phone > mobile)

File: test/support/page-assertions.js [hunk 2]
```diff
@@ -28,5 +28,21 @@ assert.textIsVisible = async function(driver, text) {
...
+assert.linkIsVisible = async function(driver, title, expectedTarget) {
+    var link = await driver.findElement(By.xpath(
+        `//a[normalize-space(.) = normalize-space('${title}')]`
+    ));
+    var visible = await link.isDisplayed();
+    var href = await link.getAttribute('href');
+
+    console.log(href);
+    console.log(expectedTarget);
+    console.log(visible);
```
Comment (danni) [inline test/support/page-assertions.js]: Left over logging.

File: test/support/page-assertions.js [hunk 3]
```diff
@@ -28,5 +28,21 @@ assert.textIsVisible = async function(driver, text) {
...
+
+    assert(visible, `Link '${title}' was present but not visible`);
```
Comment (danni) [inline test/support/page-assertions.js:47]: Apply test before making more Selenium requests. Fail fast.

File: test/support/page-assertions.js [hunk 4]
```diff
@@ -28,5 +28,21 @@ assert.textIsVisible = async function(driver, text) {
     assert(visible, `Text ${text} was present but not visible`);
 };
 
-export default assert;
+// flow:disable
+// jscs: disable
+assert.linkIsVisible = async function(driver, title, expectedTarget) {
```
Comment (danni) [inline test/support/page-assertions.js]: Flow types?

File: test/steps/servicepage.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import dictionary from "../support/dictionary";
+import unpromisify from "../support/yadda-promise";
+
+async function asyncFilter(arr, check) {
+    var results = arr.map(check);
+    results = await Promise.all(results);
+    return _.zip(results, arr)
+        .filter((e) => e[0])
+        .map((e) => e[1]);
```
Comment (danni) [inline test/steps/servicepage.js:18]: You can do this in a single map and a comprehension (comprehension syntax may be wrong).

```
var results = arr.map(element => [element, check(element)]);
return [for (element, check) of results if (check) element];
```

File: src/components/Address.js [hunk 6]
```diff
@@ -9,46 +9,50 @@ class Address extends React.Component {
     // flow:disable not supported yet
     static sampleProps = fixtures.ixa.location;
 
-    describe(property: string, description: string): React.Element {
-        var described = "";
+    describe(property: string, description: string): string {
```
Comment (danni) [inline src/components/Address.js:17]: Add a docstring to this method? I thought it was related to `object.description` and got confused.

**Comment by danni** [comment]
Pretty ace. Just a few things :)

File: test/steps/servicepage.js (hunk 5 above)
Comment (DanielHeath) [inline test/steps/servicepage.js]: Where did the `await` go?

Comment (DanielHeath) [inline test/steps/servicepage.js:18]: Where do you `await` the result of the check in that model?

Comment (danni) [inline test/steps/servicepage.js:18]: Oh, you `await Promise.all(results)`

Which is now a list of promises.

Comment (danni) [inline test/steps/servicepage.js:18]: Your closure may also have to be an async function.

Comment (DanielHeath) [inline test/steps/servicepage.js:18]: I think I've addressed everything else, but I don't see a good way to make this work more cleanly.

`results` would be a list of [element, Promise<boolean>] if I'm understanding correctly, so you couldn't use Promise.all with it

**Comment by DanielHeath** [comment]
@danni ++?

**Comment by danni** [comment]
++

**Pull request merged**